src/server/transaction.cc

```diff
@@ -1503,9 +1503,8 @@ void Transaction::FinishLogJournalOnShard(EngineShard* shard, uint32_t shard_cnt
 
 void Transaction::CancelBlocking(std::function<OpStatus(ArgSlice)> status_cb) {
   // We're on the owning thread of this transaction, so we can safely access it's data below.
-  // We still need to claim the blocking barrier, but as this function is often called blindly, we
-  // want to check first if it makes sense to even proceed.
-  if (blocking_barrier_.IsClaimed())
+  // First, check if it makes sense to proceed.
+  if (blocking_barrier_.IsClaimed() || (cid_->opt_mask() & CO::BLOCKING) == 0)
```

Comment:
I now see this error
/home/adi/dragonfly/src/server/transaction.cc:1507:55: runtime error: member access within null pointer of type 'const struct CommandId'
/home/adi/dragonfly/src/server/transaction.cc:1507:55: runtime error: member call on null pointer of type 'const struct CommandId'

when runnning memtier with eval